datadog-opentelemetry/src/ddtrace_transform.rs

```diff
@@ -49,11 +49,44 @@ pub fn otel_trace_chunk_to_dd_trace_chunk(
                 otel_resource,
             );
             otel_sampling_to_dd_sampling(trace_flags, &mut dd_span);
-            if dd_span.service == datadog_opentelemetry_mappings::DEFAULT_OTLP_SERVICE_NAME {
-                dd_span.service = BytesString::from_string(cfg.service().to_string());
-            }
+
+            add_config_metadata(&mut dd_span, cfg);
```

Comment:
Is the work in `add_config_metadata` repeated for every span? Can't we do the cfg splitting and checking once, and just add things in the loop?

Reply:
Simple case implemented